src/Compilers/Core/Portable/Serialization/StreamObjectWriter.cs

```diff
@@ -1445,16 +1483,16 @@ internal enum VariantKind
             BoxedEnum,
             DateTime,
             Array,
-            Type
+            Type,
         }
 
         internal struct Variant
         {
             public readonly VariantKind Kind;
-            private readonly decimal _image;
```

Comment:
Decimal is 128bits.  Copying these Variant structs around was expensive.  For perf, we go down to just a long, and we store decimals in the _instance field.